src/modular/boxed_residue.rs

```diff
@@ -233,6 +234,18 @@ impl BoxedResidue {
         debug_assert!(self.montgomery_form < self.residue_params.modulus);
         self.montgomery_form.clone()
     }
+
+    /// Performs the modular division by 2, that is for given `x` returns `y`
+    /// such that `y * 2 = x mod p`. This means:
+    /// - if `x` is even, returns `x / 2`,
+    /// - if `x` is odd, returns `(x + p) / 2`
+    ///   (since the modulus `p` in Montgomery form is always odd, this divides entirely).
+    pub fn div_by_2(&self) -> Self {
+        Self {
+            montgomery_form: div_by_2(&self.montgomery_form, &self.residue_params.modulus),
```

Comment:
This is a little confusing due to the name clash which makes it almost look like it's going to recurse, although it's hard to suggest something better

Reply:
What about using fully qualified namespace like `div_by_2::boxed::div_by_2`?

Reply:
Sure, sounds good